matrix-sdk-android/src/test/java/org/matrix/android/sdk/internal/session/space/DefaultResolveSpaceInfoTaskTest.kt

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2021 The Matrix.org Foundation C.I.C.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.matrix.android.sdk.internal.session.space
+
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.runBlockingTest
+import org.amshove.kluent.shouldBeEqualTo
+import org.junit.Test
+import org.matrix.android.sdk.test.fakes.FakeGlobalErrorReceiver
+import org.matrix.android.sdk.test.fakes.FakeSpaceApi
+
+@ExperimentalCoroutinesApi
+class DefaultResolveSpaceInfoTaskTest {
+
+    private val spaceApi = FakeSpaceApi()
+    private val globalErrorReceiver = FakeGlobalErrorReceiver()
+    private val resolveSpaceInfoTask = DefaultResolveSpaceInfoTask(spaceApi.instance, globalErrorReceiver)
+
+    @Test
+    fun `given stable endpoint works, when execute, then return stable api data`() = runBlockingTest {
+        spaceApi.givenStableEndpointWorks()
```

Comment:
not a strong opinion just a personal preference, it can be helpful from the test standpoint to be able to relate the inputs and results

```kotlin
// give stable endpoint returns a response, when execute, 
spaceApi.givenStableEndpointReturns(A_RESPONSE)

val result = resolveSpaceInfoTask.execute(spaceApi.params)

result shouldBeEqualTo A_RESPONSE
```

I'm wondering there's a way to relate to get the best of both worlds (with the boilerplate reduction and tightening the relationship between the given and the expected result)

```kotlin
fakeSpaceApi.givenStableEndpointSuccess()

val result = resolveSpaceInfoTask.execute(spaceApi.params)

result shouldBeEqualTo fakeSpaceApi.successResponse
```

what do you think?

Reply:
Tbh I like the first option as it gives the test class more flexibility about the types of responses it can test with but still maintains a very readable front. I'll go with that!